modin/experimental/engines/omnisci_on_ray/frame/df_algebra.py

```diff
@@ -0,0 +1,351 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import abc
+from .expr import InputRefExpr
+from collections import OrderedDict
+
+
+class TransformMapper:
+    def __init__(self, op):
+        self._op = op
+
+    def translate(self, col):
+        if col == "__rowid__":
+            return self._op.input[0].ref(col)
+        return self._op.exprs[col]
+
+
+class FrameMapper:
+    def __init__(self, frame):
+        self._frame = frame
+
+    def translate(self, col):
+        return self._frame.ref(col)
+
+
+class InputMapper:
+    def __init__(self):
+        self._mappers = {}
+
+    def add_mapper(self, frame, mapper):
+        self._mappers[frame] = mapper
+
+    def translate(self, ref):
+        if ref.modin_frame in self._mappers:
+            return self._mappers[ref.modin_frame].translate(ref.column)
+        return ref
+
+
+class DFAlgNode(abc.ABC):
+    """Base class for all DataFrame Algebra nodes"""
+
+    @abc.abstractmethod
+    def copy(self):
+        pass
+
+    def walk_dfs(self, cb, *args, **kwargs):
+        if hasattr(self, "input"):
+            for i in self.input:
+                i._op.walk_dfs(cb, *args, **kwargs)
+        cb(self, *args, **kwargs)
+
+    def collect_partitions(self):
+        partitions = []
+        self.walk_dfs(lambda a, b: a._append_partitions(b), partitions)
+        return partitions
+
+    def collect_frames(self):
+        frames = []
+        self.walk_dfs(lambda a, b: a._append_frames(b), frames)
+        return frames
+
+    def _append_partitions(self, partitions):
+        pass
+
+    def _append_frames(self, frames):
+        pass
+
+    def __repr__(self):
+        return self.dumps()
+
+    def dump(self, prefix=""):
+        print(self.dumps(prefix))
```

Comment:
I prefer the flake8 `noqa` for `T001` to be added to this line rather than the whole file.

Reply:
fixed